packages/material-renderers/src/complex/MaterialAnyOfRenderer.tsx

```diff
@@ -46,12 +50,41 @@ export const MaterialAnyOfRenderer = ({
   cells,
   uischema,
   uischemas,
-}: StatePropsOfCombinator) => {
+  id,
+  data,
+}: CombinatorRendererProps) => {
   const [selectedAnyOf, setSelectedAnyOf] = useState(indexOfFittingSchema || 0);
-  const handleChange = useCallback(
-    (_ev: any, value: number) => setSelectedAnyOf(value),
-    [setSelectedAnyOf]
+  const [open, setOpen] = useState(false);
+  const [newSelectedIndex, setNewSelectedIndex] = useState(0);
+
+  const handleClose = useCallback(() => setOpen(false), [setOpen]);
+
+  const handleTabChange = useCallback(
+    (_event: any, newIndex: number) => {
+      if (
+        isEmpty(data) ||
+        typeof data ===
+          typeof createDefaultValue(anyOfRenderInfos[newIndex].schema)
```

Comment:
What is the purpose of this? Is this to avoid unnecessary confirmations in case the data is "the same"?
If yes, I see a risk of skipping the  confirmation if two anyOfs have the same data type. This could easily happen if they are both objects.
You can test this in the deployed example at https://deploy-preview-2175--jsonforms-examples.netlify.app/react-material/#anyOf
When you change some data and then switch the tab, no confirm dialog is opened.

Reply:
The purpose was to fix #2173, where unfitting data types would lead to a crash. For example, if you switch between an object and an array. I would argue that the current behaviour is correct. AnyOf means that several of the subschemas can be fulfilled. Resetting the data every time would represent the oneOf option.

Reply:
That makes sense. Thanks for the explanation :)